<a href="https://colab.research.google.com/github/puckgroen/TM10007_PROJECT/blob/master/assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# TM10007 Assignment template

In [1]:
# Run this to use from colab environment
!pip install -q --upgrade git+https://github.com/karinvangarderen/tm10007_project.git

In [0]:
# Download the necessary packages
import numpy as np

# Classifiers
from sklearn import model_selection

## Data loading and cleaning

Below are functions to load the dataset of your choice. After that, it is all up to you to create and evaluate a classification method. Beware, there may be missing values in these datasets. Good luck!

In [12]:
# Data loading functions. Uncomment the one you want to use

from hn.load_data import load_data

data = load_data()
print(f'The number of samples: {len(data.index)}')
print(f'The number of features: {len(data.columns)}')
print(data.keys())
Y = data['label']
print(Y)

The number of samples: 113
The number of features: 160
Index(['hf_energy', 'hf_entropy', 'hf_kurtosis', 'hf_max', 'hf_mean',
       'hf_median', 'hf_min', 'hf_peak', 'hf_quartile_range', 'hf_range',
       ...
       'tf_LBP_skew_R3_P12', 'tf_LBP_skew_R8_P24', 'tf_LBP_std_R15_P36',
       'tf_LBP_std_R3_P12', 'tf_LBP_std_R8_P24', 'tf_NGTDM_Busyness',
       'tf_NGTDM_Coarseness', 'tf_NGTDM_Complexity', 'tf_NGTDM_Contrast',
       'tf_NGTDM_Strength'],
      dtype='object', length=160)


The data has 160 features and 113 samples/subjects. The labels are given as either T12 (low) or T34 (high). 

The aim of this study is to predict the T-stage (high/low) in patients with H&N cancer based on features, extracted from CT. A good performance on this dataset would be above 70% mean accuracy.

In [30]:
# Split the dataset in design and test set. Later on the design set should make use of cross-validation? 
X_train, X_test, y_train, y_test = model_selection.train_test_split(data, Y, test_size=0.2, stratify=Y)
print(len(X_test))
print(len(X_train))


# Continue with the train dataset and use a 4 fold cross-validation 
cv_4fold = model_selection.StratifiedKFold(n_splits=4)


# Loop over the folds
for validation_index, test_index in cv_4fold.split(X_train, y_train):
    # For now I had to rearrange the dictionary of X_train to a list, otherwise we cannot index it. 
    # Not sure if this is the perfect way but just to make it visual, you can see the validation sizes printed. This seems to be ok!
    print('Validation size in current fold =')
    array = np.array(list(X_train.items()))
    
    # Split the data properly
    X_validation = array[validation_index]
    
    print(len(X_validation))
    y_validation = y_train[validation_index]
    
    X_test = array[test_index]
    y_test = y_train[test_index]

23
90
Validation size in current fold =
67
Validation size in current fold =
67
Validation size in current fold =
68
Validation size in current fold =
68
